code/go/0chain.net/miner/miner/main.go

```diff
@@ -204,7 +204,28 @@ func main() {
 	initIntegrationsTests(node.Self.Underlying().GetKey())
 
 	var server *http.Server
+	var profServer *http.Server
+
+	mux := http.NewServeMux()
+	http.DefaultServeMux = mux
```

Comment:
Prof is pre-initialized for default server mux. If we do not replace this, prof handlers will be available for main router, which is not expected behavior. I have checked it myself in a test setting.

Reply:
Ah, that makes sense. Then we can make the switch code simpler, i.e replace the default mux with new one for disabling the pprof, and use the default router when enable the pprof. In this way, we don't need to setup pprof handlers manually, also we don't need to change the port number. So the changes we need for switching is alike:

```go
if !enabelPProf() {
	http.DefaultServeMux = http.NewServeMux()
}
```

Reply:
I have already had this discussion with @dabasov. We dont want pprof on the main router ever, so that someone doesn't enable it in production by mistake. So pprof will be switched on an off on a different router running on a different port.

Reply:
Enable it on main router is not a problem, we may enable it to do [Continuous Profiling](https://medium.com/google-cloud/continuous-profiling-of-go-programs-96d4416af77b) in the future, and it should be safe to enable pprof in prod. The main point is to make the change/code simple, and avoid unnecessary changes on both code logic and deployment scripts. This change will fail the pprof on networks if we don't update the helm repo.

Reply:
all debug functionality should be removed to dev, it makes no sense for end user and regular miner will never switch on profiling, no matter how much resources is spent, simply no use-case here.Devs can always switch on it for devv networks, good separation of concerns, I don't see any reason to keep it in the same router

Reply:
It doesn't make any difference actually. We can disable the pprof in production, and enable it in dev only. This is a simple feature, which only require 3 lines of code changes. But running it in another router would require changes in both the code, deployment scripts and 0helm config, and I actually don't see much benefit of doing it. Anyway, if you insist to do this, please remember to update 0helm to add a new ngix ingress router for pprof.

Reply:
@peterlimg I'm afraid that it will create the overhead of profiling functions, it might be not big, but nevertheless meaningful

Reply:
Sure, let's merge this one and update the 0helm repo.